adapters/nexx360/nexx360.go

```diff
@@ -0,0 +1,236 @@
+	Caller []Nexx360Caller `json:"caller,omitempty"`
+}
+
+type Nexx360ResBidExt struct {
+	BidType string `json:"bidType,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{endpoint: config.Endpoint}, nil
+}
+
+// CALLER Info used to track Prebid Server
+// as one of the hops in the request to exchange
+
+func getVersion() string {
+	if version.Ver != "" {
+		return version.Ver
+	}
+	return "n/a"
+}
+
+var CALLER = Nexx360Caller{"Prebid-Server", getVersion()}
+
+func processImps(impList []openrtb2.Imp) (imp []openrtb2.Imp, tagId string, placement string, error error) {
+	var imps []openrtb2.Imp
+	for idx, imp := range impList {
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var nexx360Ext openrtb_ext.ExtImpNexx360
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &nexx360Ext); err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		impExt := Ext{
+			Nexx360: bidderExt.Bidder,
+		}
+
+		impExtJSON, err := json.Marshal(impExt)
+		if err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		imp.Ext = impExtJSON
+		imps = append(imps, imp)
+		if idx == 0 {
+			var nexx360Ext openrtb_ext.ExtImpNexx360
+			if err := jsonutil.Unmarshal(bidderExt.Bidder, &nexx360Ext); err != nil {
+				return nil, "", "", &errortypes.BadInput{
+					Message: err.Error(),
+				}
+			}
+			tagId = nexx360Ext.TagId
+			placement = nexx360Ext.Placement
+		}
+
+	}
+
+	return imps, tagId, placement, nil
+}
+
+func makeReqExt() ([]byte, error) {
+	reqExt := ReqExt{
+		Nexx360: &ReqNexx360Ext{
+			Caller: []Nexx360Caller{CALLER},
+		},
+	}
+
+	return json.Marshal(reqExt)
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var imp, tagId, placement, err = processImps(request.Imp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	request.Imp = imp
+
+	urlBuilder, err := url.Parse(a.endpoint)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	query := url.Values{}
+
+	const placementKey = "placement"
+	if placement != "" {
+		query[placementKey] = []string{placement}
+	}
+
+	const tagIdKey = "tag_id"
+	if tagId != "" {
+		query[tagIdKey] = []string{tagId}
+	}
+	urlBuilder.RawQuery = query.Encode()
+
+	uri := urlBuilder.String()
+
+	reqExt, err := makeReqExt()
+	if err != nil {
+		return nil, []error{err}
+	}
+	request.Ext = reqExt
+
+	// Last Step
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+	fmt.Printf("Nexx360: Request JSON: %s\n", string(reqJSON))
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+
+	adapter := &adapters.RequestData{
+		Method:  "POST",
```

Comment:
Use `http.MethodPost` instead of `"POST"`.

Reply:
Done